packages/itwinui-react/src/core/Table/types/react-table-types.d.ts

```diff
@@ -0,0 +1,1320 @@
+  canSort: boolean;
+  toggleSortBy: (descending?: boolean, multi?: boolean) => void;
+  getSortByToggleProps: (
+    props?: Partial<TableSortByToggleProps>,
+  ) => TableSortByToggleProps;
+  clearSortBy: () => void;
+  isSorted: boolean;
+  sortedIndex: number;
+  isSortedDesc: boolean | undefined;
+}
+
+export type OrderByFn<D extends Record<string, unknown>> = (
+  rowA: Row<D>,
+  rowB: Row<D>,
+) => number;
+export type SortByFn<D extends Record<string, unknown>> = (
+  rowA: Row<D>,
+  rowB: Row<D>,
+  columnId: IdType<D>,
+  desc?: boolean,
+) => number;
+
+export type DefaultSortTypes =
+  | 'alphanumeric'
+  | 'datetime'
+  | 'basic'
+  | 'string'
+  | 'number';
+
+export interface SortingRule<D> {
+  id: IdType<D>;
+  desc?: boolean | undefined;
+}
+
+//#endregion
+
+// Additional API
+export const actions: Record<string, string>;
+export type ActionType = { type: string } & Record<string, any>;
+export const defaultColumn: Partial<Column> & Record<string, any>;
+
+// Helpers
+export type StringKey<D> = Extract<keyof D, string>;
+export type IdType<D> = StringKey<D> | string;
+export type CellValue<V = any> = V;
+
+export type Renderer<Props> =
+  | ComponentType<Props>
+  | ReactElement
+  | string
+  | number
+  | ReactFragment;
+
+export interface PluginHook<D extends Record<string, unknown>> {
+  (hooks: Hooks<D>): void;
+  pluginName?: string | undefined;
+}
+
+export type TableDispatch<A = any> = (action: A) => void;
+
+// utils
+export function defaultOrderByFn<D extends Record<string, unknown> = {}>(
+  arr: Array<Row<D>>,
+  funcs: Array<OrderByFn<D>>,
+  dirs: boolean[],
+): Array<Row<D>>;
+
+export function defaultGroupByFn<D extends Record<string, unknown> = {}>(
+  rows: Array<Row<D>>,
+  columnId: IdType<D>,
+): Record<string, Array<Row<D>>>;
+
+export function makePropGetter(hooks: Hooks, ...meta: any[]): any;
+
+export function reduceHooks<T extends object = {}>(
+  hooks: Hooks,
+  initial: T,
+  ...args: any[]
+): T;
+
+export function loopHooks(hooks: Hooks, ...args: any[]): void;
+
+export function ensurePluginOrder<D extends Record<string, unknown> = {}>(
+  plugins: Array<PluginHook<D>>,
+  befores: string[],
+  pluginName: string,
+): void;
+
+export function functionalUpdate<D extends Record<string, unknown> = {}>(
+  updater: any,
+  old: Partial<TableState<D>>,
+): Partial<TableState<D>>;
+
+export function useGetLatest<T>(obj: T): () => T;
+
+export function safeUseLayoutEffect(
+  effect: EffectCallback,
+  deps?: DependencyList,
+): void;
+
+export function useMountedLayoutEffect(
+  effect: EffectCallback,
+  deps?: DependencyList,
+): void;
+
+export function useAsyncDebounce<F extends (...args: any[]) => any>(
+  defaultFn: F,
+  defaultWait?: number,
+): F;
+
+export function makeRenderer(
+  instance: TableInstance,
+  column: ColumnInstance,
+  meta?: any,
+): ReactElement;
+
+//////////////////////////////////////////////////////////////////////////////////////////////////
```

Comment:
nit: I think we are using `//-----------------` as separators.

Reply:
https://github.com/iTwin/iTwinUI/pull/1514/commits/61137eae6d0aae0e6d174f99727859c26d63b6e0: Changed